packages/amplify-graphql-api-construct-tests/src/utils/sql-crudl-helper.ts

```diff
@@ -1,6 +1,16 @@
 import AWSAppSyncClient from 'aws-appsync';
 import gql from 'graphql-tag';
 
+/**
+ * Type that represents the GraphQL field selection string structure
+ * GraphQL does not diffrentiate between return single object or array of primitives/objects
```

Comment:
nit: This is unclear. GraphQL definitely allows you to specify arrays distinctly from custom types. Do you mean that the selection set syntax is just a list of fields? Add an example to show what this structure is intended to solve and how it should be used. `FieldMap` is generic enough that it's not clear from the type name what the right place to use it is

Reply:
More comments and explanations added for `FieldMap`